### PR TITLE
Use evm for TravisCI testing

### PR DESCRIPTION
**Pull request by xiongtx**
Per @bbatsov's [suggestion](https://github.com/clojure-emacs/cider/pull/2059#issuecomment-317231168), we'd like to align TravisCI tests for `clojure-mode` with those of CIDER.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,22 +1,24 @@
...
+  - EMACS_BINARY=emacs-24.4-travis MAKE_TEST=test
+  - EMACS_BINARY=emacs-24.4-travis MAKE_TEST=test-bytecomp
+  - EMACS_BINARY=emacs-24.5-travis MAKE_TEST=test
+  - EMACS_BINARY=emacs-24.5-travis MAKE_TEST=test-bytecomp
+  - EMACS_BINARY=emacs-25.1-travis MAKE_TEST=test
+  - EMACS_BINARY=emacs-25.1-travis MAKE_TEST=test-bytecomp
+  - EMACS_BINARY=emacs-25.2-travis MAKE_TEST=test
+  - EMACS_BINARY=emacs-25.2-travis MAKE_TEST=test-bytecomp
+  - EMACS_BINARY=emacs-25.2-travis MAKE_TEST=test-checks
+  - EMACS_BINARY=emacs-git-snapshot-travis
```
Comment (vspinu) [inline .travis.yml]: Is such a big matrix really needed? How about saving some trees and unifying all tests per emacs version?

Comment (xiongtx) [inline .travis.yml]: Separating the tests makes sense--I like dealing w/ style issues separately from actual test issues.

**Comment by xiongtx** [comment]
Ping @bbatsov.

**Comment by bbatsov** [comment]
Excellent work! 👍

**Pull request merged**